NumberTheory/FastPower.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+<Fast Power>
+--------------
+Fast Power is an optimized algorithm to compute exponentiation in a short time.
+
+
+Time Complexity
+-----------------
+O(log(N)) where N is the power the number is raised to.
+
+Space Complexity
+------------------
+O(log(N)) where N is the power the number is raised to.
```

Comment:
~`where N is the power the number is raised to`~
More simply, `where N is the exponent.`